concent_api/core/views.py

```diff
@@ -43,15 +46,33 @@ def send(_request, client_message):
             golem_message,
             message_timestamp,
         )
+        store_message_auth(
+            golem_message,
+            provider_public_key  = client_public_key,
+            requestor_public_key = requestor_public_key,
```

Comment:
You're inserting `requestor_public_key` into a non-nullable column but it can be `None` (`decode_requestor_public_key()` may return `None`) and you do not have any checks against this condition here.

Reply:
I have changed it to catch KeyError and raise Http400 when requestor key is missing in header but expected to be there.

Reply:
Please don't catch `KeyError`. Check it with an `if`.

Reply:
:+1: